pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordReaderFactory.java

```diff
@@ -0,0 +1,203 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.utils.JsonUtils;
+
+
+public class RecordReaderFactory {
+
+  private static final Map<String, String> DEFAULT_RECORD_READER_CLASS_MAP = new HashMap<>();
+  private static final Map<String, String> DEFAULT_RECORD_READER_CONFIG_CLASS_MAP = new HashMap<>();
+
+  // TODO: This could be removed once we have dynamic loading plugins supports.
+  private static final String DEFAULT_AVRO_RECORD_READER_CLASS = "org.apache.pinot.avro.data.readers.AvroRecordReader";
+  private static final String DEFAULT_CSV_RECORD_READER_CLASS = "org.apache.pinot.csv.data.readers.CSVRecordReader";
+  private static final String DEFAULT_CSV_RECORD_READER_CONFIG_CLASS =
+      "org.apache.pinot.csv.data.readers.CSVRecordReaderConfig";
+  private static final String DEFAULT_JSON_RECORD_READER_CLASS = "org.apache.pinot.json.data.readers.JSONRecordReader";
+  private static final String DEFAULT_THRIFT_RECORD_READER_CLASS =
+      "org.apache.pinot.thrift.data.readers.ThriftRecordReader";
+  private static final String DEFAULT_THRIFT_RECORD_READER_CONFIG_CLASS =
+      "org.apache.pinot.thrift.data.readers.ThriftRecordReaderConfig";
+  private static final String DEFAULT_ORC_RECORD_READER_CLASS = "org.apache.pinot.orc.data.readers.ORCRecordReader";
+  private static final String DEFAULT_PARQUET_RECORD_READER_CLASS =
+      "org.apache.pinot.parquet.data.readers.ParquetRecordReader";
+
+  private RecordReaderFactory() {
+  }
+
+  public static void register(String fileFormat, String recordReaderClassName, String recordReaderConfigClassName) {
+    DEFAULT_RECORD_READER_CLASS_MAP.put(fileFormat.toUpperCase(), recordReaderClassName);
+    DEFAULT_RECORD_READER_CONFIG_CLASS_MAP.put(fileFormat.toUpperCase(), recordReaderConfigClassName);
+  }
+
+  public static void register(FileFormat fileFormat, String recordReaderClassName, String recordReaderConfigClassName) {
+    register(fileFormat.name(), recordReaderClassName, recordReaderConfigClassName);
+  }
+
+  /**
+   * Construct a RecordReaderConfig instance from a given file path.
+   *
+   * @param recordReaderConfigClassName
+   * @param readerConfigFile
+   * @return
+   * @throws IOException
+   * @throws ClassNotFoundException
+   */
+  public static RecordReaderConfig getRecordReaderConfigByClassName(String recordReaderConfigClassName, String readerConfigFile)
+      throws IOException, ClassNotFoundException {
+    return getRecordReaderConfigByClassName(recordReaderConfigClassName, new File(readerConfigFile));
+  }
+
+  /**
+   * Construct a RecordReaderConfig instance from a given file.
+   *
+   * @param recordReaderConfigClassName
+   * @param readerConfigFile
+   * @return
+   * @throws IOException
+   * @throws ClassNotFoundException
+   */
+  public static RecordReaderConfig getRecordReaderConfigByClassName(String recordReaderConfigClassName, File readerConfigFile)
```

Comment:
do we need this method? Can we simplify RecordReaderConfig class-related methods. can we initialize the RecordReader with a simple properties/map object

Another option is get rid of RecordReaderConfig completely and pass Map<String, Object> in RecordReader init method.